libs/island-ui/core/src/lib/DropdownMenu/useMenuHoverProps.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { MenuStateReturn } from 'reakit/Menu'
+
+interface HoverPropsRefState {
+  hoverCount: number
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  hideTimeout?: any
+}
+
+const MENU_LEAVE_DELAY = 500
+export const useMenuHoverProps = (
+  menu: MenuStateReturn,
+  openOnHover: boolean,
+) => {
+  const hoverRef = React.useRef<HoverPropsRefState>({
+    hoverCount: 0,
+    hideTimeout: undefined,
+  })
+  const hoverState = hoverRef.current
+
+  return openOnHover
+    ? {
+        onMouseEnter: () => {
+          hoverState.hoverCount++
```

Comment:
Yeah, it's defensive code since these event handlers is added to two elements and I don't want to encounter issues with event ordering now or later.

Also found a way to fix the page scrolling.

Reply:
Cool